rinad/src/ipcm/app-handlers.cc

```diff
@@ -47,61 +47,68 @@ void IPCManager_::os_process_finalized_handler(
 			<< "terminated" << endl;
 	FLUSH_LOG(INFO, ss);
 
-	//Prevent any insertion/deletion to happen
-	rina::ReadScopedLock readlock(ipcp_factory_.rwlock);
-
-	// Look if the terminating application has allocated flows
-	// with some IPC processes
-	collect_flows_by_application(app_name, involved_flows);
-	unsigned short ipcp_id = 0;
-	for (list<rina::FlowInformation>::iterator fit = involved_flows.begin();
-			fit != involved_flows.end(); fit++) {
-
-		IPCMIPCProcess *ipcp = select_ipcp_by_dif(fit->difName);
-		if (!ipcp) {
-			ss  << ": Cannot find the IPC process "
-					"that provides the flow with port-id " <<
-					fit->portId << endl;
-			FLUSH_LOG(ERR, ss);
-			continue;
-		}
+	{
+		//Prevent any insertion/deletion to happen
+		rina::ReadScopedLock readlock(ipcp_factory_.rwlock);
+
+		// Look if the terminating application has allocated flows
+		// with some IPC processes
+		collect_flows_by_application(app_name, involved_flows);
+		unsigned short ipcp_id = 0;
+		for (list<rina::FlowInformation>::iterator fit = involved_flows.begin();
+				fit != involved_flows.end(); fit++) {
+
+			IPCMIPCProcess *ipcp = select_ipcp_by_dif(fit->difName);
+			if (!ipcp) {
+				ss  << ": Cannot find the IPC process "
+						"that provides the flow with port-id " <<
+						fit->portId << endl;
+				FLUSH_LOG(ERR, ss);
+				continue;
+			}
 
-		{
-			//Auto release the read lock
-			rina::ReadScopedLock readlock(ipcp->rwlock, false);
-			ipcp_id = ipcp->get_id();
-		}
+			{
+				//Auto release the read lock
+				rina::ReadScopedLock readlock(ipcp->rwlock, false);
+				ipcp_id = ipcp->get_id();
```

Comment:
This lock idiom is not correct - in general. You assign the local variable under the lock, but then you use it outside the lock (line 77): when you use it, the ipcp->get_id() value may have already been changed under your feet. Is that what you want? I know that the ipcp id should not change, but then why you need a lock to access the ipcp id?